src/CartesianProduct.jl

```diff
@@ -562,3 +563,147 @@ function same_block_structure(x::AbstractVector{S1}, y::AbstractVector{S2}
     end
     return true
 end
+
+"""
+    block_structure(cpa::CartesianProductArray{N}) where {N}
+
+Returns an array containing the dimension ranges of each block in a `CartesianProductArray`.
+
+### Input
+
+- `cpa` -- Cartesian product array
+
+### Output
+
+A vector of ranges
+
+### Example
+
+```jldoctest
+julia> using LazySets: block_structure
```

Comment:
Just a comment: Instead you can write `LazySets.block_structure` below (as you prefer).